hypothesis/client
# Host toast messages

This PR extends what was implemented in https://github.com/hypothesis/client/pull/5342, and makes use of the new RPC messages to render a new `aria-live` region in the host frame that can render toast messages forwarded from the sidebar while collapsed.

In order to do it, the existing `ToastMessages` component has been split in two:
* `BaseToastMessages`: It's a controlled component, which has 95% of the original code, but expects the `messages` to be provided as props and does not depend on neither the sidebar store, or the `ToastMessengerService`
* `sidebar/components/ToastMessages`: The "original" component, which depends on the sidebar store and the `ToastMessengerService`, and wraps a `BaseToastMessages` component.

Then, a new `annotator/components/ToastMessages` component has been created, also wrapping the `BaseToastMessages`, but listening for the `toastMessagePushed` and `toastMessageDismissed` RPC messages from `sidebarRPC` in order to add/remove messages.

### Testing steps:

1. Check out this branch
2. Run a screen reader
3. With the sidebar expanded, create a new highlight (either via mouse click or keyboard shortcut) -> The screen reader should announce "Success: Highlight saved" only once.
4. Collapse the sidebar.
5. Create another highlight (either via mouse click or keyboard shortcut) -> The screen reader should announce "Success: Highlight saved" only once.
